### PR TITLE
[AIRFLOW-3628] Add smtp_mime_from

### PR DESCRIPTION
**Pull request by yujiantao**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3628
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:
In airflow alert email, the value of the from field is the source email address, but sometimes we want to show the receiver where the alert is really from. For example, the email was sent from no-reply@example.com, but we want to show the receiver it was from "airflow", so that it is easy to distinguish it from other messages. With this PR, we just need to set "smtp_mime_from = "airflow" in airflow.cfg.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -342,7 +342,7 @@ smtp_ssl = False
 # smtp_password = airflow
 smtp_port = 25
 smtp_mail_from = airflow@example.com
-
+smtp_mime_from = airflow
```
Comment (feng-tao) [inline airflow/config_templates/default_airflow.cfg:345]: not sure if I understand, but what is the difference between ``smtp_mail_from`` and ``smtp_mime_from`` in this case? If we put smtp_mail_from = airflow, could we achieve what you are trying to do?

Comment (yujiantao) [reply airflow/config_templates/default_airflow.cfg:345]: 1. smtp_mail_from must be an email address from where the alert email is going to be sent. if we put smtp_mail_from = airflow, the email sending will fail. 
2. if we set msg['From'] = smtp_mail_from, the receiver will see the email sender is the value fo smtp_mail_from which is an email address. Sometimes the the email address is shared by multiple services like airflow, monitor, data quality, etc. If one received email messages from all these services, it is impossible to distinguish messages by the email sender field quickly. But if we set an alias for the email sender by setting  smtp_mime_from = airflow and msg['From'] = smtp_mime_from, the receiver will see the email from the alias and it is easy to figure out the email is from airflow or other services.

Comment (feng-tao) [reply airflow/config_templates/default_airflow.cfg:345]: @yujiantao , thank for the explanation.  So the issue is that if we use a single alert email address  to send info for every services(airflow, monitor, etc), we can't tell this email from subject where it comes from. Hence we add a note(smtp_mime_from) to indicate the subject line.

Comment (feng-tao) [reply airflow/config_templates/default_airflow.cfg:345]: So it is possible to add a test for this new config?

Comment (yujiantao) [reply airflow/config_templates/default_airflow.cfg:345]: @feng-tao commit update including test

**Comment by OmerJog** [comment]
@yujiantao 
Can you fix the Flake8 violations ?
```
./tests/core.py:2748:111: E501 line too long (113 > 110 characters)
./tests/core.py:2778:111: E501 line too long (113 > 110 characters)

```

**Comment by OmerJog** [comment]
@yujiantao any progress?

**Pull request closed without merging**